src/iotile/cloud.service.ts

```diff
@@ -411,7 +411,7 @@ export class CloudService {
 
   public getStreamData(stream: Stream, args: DataFilterArgs): Observable<Array<DataPoint>>  {
 
-    let url: string = '/stream/' + stream.slug + '/data/';
+    let url: string = '/filter?=' + stream.slug;
```

Comment:
This won't work as you will add double `?` characters if any args are passed. The URL is also wrong, as it needs to be '/data/`
Keep the url as `/data/
and add the filter to the args object. Something like

```
 let url: string = '/data/';
args.filter = stream.slug
```

(but you will need to enhance `DataFilterArgs` to support filter